libcontainer/configs/validate/rootless.go

```diff
@@ -0,0 +1,100 @@
+package validate
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+type RootlessValidator struct {
+}
+
+func (v *RootlessValidator) Validate(config *configs.Config) error {
+	// There's nothing to validate.
+	if config == nil {
+		return nil
+	}
+
+	if err := v.mount(config); err != nil {
+		return err
+	}
+	// If we're running under Linux with CLONE_NEWCGROUP, we don't have to worry
+	// about any of this.
+	if !config.Namespaces.Contains(configs.NEWCGROUP) {
+		if err := v.cgroup(config); err != nil {
+			return err
+		}
+	}
+
+	// XXX: We currently can't verify the user config at all, because
+	//      configs.Config doesn't store the user-related configs. So this
+	//      has to be verified by setupUser() in init_linux.go.
+
+	return nil
+}
+
+// cgroup verifies that the user isn't trying to set any cgroup limits or paths.
+func (v *RootlessValidator) cgroup(config *configs.Config) error {
+	// Nothing set at all.
+	if config.Cgroups == nil || config.Cgroups.Resources == nil {
+		return nil
+	}
+
+	// Used for comparison.
```

Comment:
runc should populate the correct config that libcontainer gets and it should not be modified inside libcontainer.  All the changes that need to be made should happen while we generate the config not after it is made.